### PR TITLE
Tune element UID generation

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
The idea is based on https://github.com/facebook/WebDriverAgent/pull/897/ by @qmfrederik. The change allows to nicely optimise the size of the internal cache.

### CODE REVIEW

File: WebDriverAgentLib/Routing/FBElementUtils.m [hunk 1]
```diff
@@ -111,15 +111,23 @@ + (NSString *)wdAttributeNameForAttributeName:(NSString *)name
...
-    return [[element.payload objectForKey:@"uid.elementID"] intValue];
+    elementId = [[element.payload objectForKey:@"uid.elementID"] longLongValue];
+  } else {
+    elementId = [[element valueForKey:@"_elementID"] longLongValue];
   }
-  return [[element valueForKey:@"_elementID"] intValue];
+  int processId = element.processIdentifier;
+  uint8_t b[16] = {0};
+  memcpy(b, &elementId, sizeof(long long));
+  memcpy(b + sizeof(long long), &processId, sizeof(int));
```
Comment (qmfrederik) [inline WebDriverAgentLib/Routing/FBElementUtils.m:128]: I was thinking about this yesterday; perhaps you want to assert that `sizeof(long long) + sizeof(int) < 16`, although that's guaranteed to be the case on 32-bit and 64-bit platforms.

**Comment by qmfrederik** [comment]
Looks good to me - thanks for picking this up!

To recap, XCUITest uses 'proxy' objects which point to the actual UI elements. Every time you query for an element, you get new proxy objects - even if they point to the same underlying UI element.

This PR changes the algorithm which assigns UUIDs to XCUIElements so that it is 'deterministic' - in that two XCUIElements which point to the same UI element will have the same UUID.

This will have a positive impact on the element cache size - querying the same element 1000 times would result in ~125MB memory consumption in the current WebDriverAgent implementation; with this fix the memory consumption is reduced to ~25MB.

Together with the ideas discussed in in facebook/WebDriverAgent#896, this should seriously reduce the memory consumption of the element cache.

**Review by imurchie — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
@qmfrederik Can you please help to update cache unit tests after merge? 

https://travis-ci.org/appium/WebDriverAgent/jobs/359731836

**Comment by mykola-mokhnach** [comment]
nwm, I've figured it out myself: https://github.com/appium/WebDriverAgent/pull/65/commits/d7061da9098afee7bb60620f954779246c35050b

**Pull request merged**